corehq/ex-submodules/casexml/apps/phone/dbaccessors/sync_logs_by_user.py

```diff
@@ -22,3 +27,39 @@ def get_all_sync_logs_docs():
         reduce=False,
     )]
     return iter_docs(SyncLog.get_db(), all_sync_log_ids)
+
+
+def get_sync_logs_for_user(user_id, limit):
+    rows = synclog_view(
+        "phone/sync_logs_by_user",
+        startkey=[user_id, {}],
+        endkey=[user_id],
+        descending=True,
+        reduce=False,
+        limit=limit,
+        include_docs=True,
+    )
+    sync_log_jsons = (row['doc'] for row in rows)
+    return [properly_wrap_sync_log(sync_log_json) for sync_log_json in sync_log_jsons]
+
+
+def synclog_view(view_name, **params):
+    return combine_views([SyncLog.get_db(), get_db(None)], view_name, **params)
+
+
+def combine_views(dbs, view_name, **params):
```

Comment:
hmm, ok. Yeah only not supporting skip because there's no way to do it even sort of efficiently. (you have to just get all the things you're "skipping", sort them, and skip them in memory; with limit it's at least a constant factor N equal to the number of dbs you're checking)